supervisor/api/ingress.py

```diff
@@ -217,11 +258,33 @@ async def _handle_request(
 
             return response
 
+    async def _find_user_by_id(self, user_id: str) -> IngressSessionDataUser | None:
+        """Find user object by the user's ID."""
+        try:
+            list_of_users = await self.sys_homeassistant.get_users()
+        except (HomeAssistantAPIError, TypeError) as err:
+            _LOGGER.error(
+                "%s error occurred while requesting list of users: %s", type(err), err
+            )
+            return None
+
+        if list_of_users is not None:
+            self._list_of_users = list_of_users
 
-def _init_header(request: web.Request, addon: str) -> CIMultiDict | dict[str, str]:
+        return next((user for user in self._list_of_users if user.id == user_id), None)
+
+
+def _init_header(
+    request: web.Request, addon: Addon, session_data: IngressSessionData | None
+) -> CIMultiDict | dict[str, str]:
     """Create initial header."""
     headers = {}
 
+    if session_data is not None:
+        headers[HEADER_REMOTE_USER_ID] = session_data.user.id
+        headers[HEADER_REMOTE_USER_NAME] = session_data.user.username
+        headers[HEADER_REMOTE_USER_DISPLAY_NAME] = session_data.user.display_name
+
```

Comment:
Can you ditch those fields from the header if not needed? Otherwise we have an security issue that we forward the information as it is from the client

Reply:
@pvizeli Could you give me a hin how you imagine this, please? Initially my PR had methods to activate this for every addon in the addon configuration. However, I was told to remove that. From the security perspective I could imagine:

* send only user ID (this obviously not secures the ID, but the user's name)
* enable sending the data per addon in the addon configuration
* do not send any information initially, but provide an API where the addon can retrieve user information by its session ID

Reply:
Oh, I see... I misunderstood the requirement :)

Reply:
Meh, I see now, we should remove it on lin 293 which included the headers from remote

Reply:
I updated it in that regard